dpnp/tests/test_indexing.py

```diff
@@ -1307,3 +1309,51 @@ def test_error(self):
             dpnp.select([x0], [x1], default=x1)
         with pytest.raises(TypeError):
             dpnp.select([x1], [x1])
+
+
+def test_compress_basic():
+    a = dpnp.arange(16).reshape(4, 4)
+    condition = dpnp.asarray([True, False, True])
+    r = dpnp.compress(condition, a, axis=0)
+    assert_array_equal(r[0], a[0])
```

Comment:
Usually we compare result against numpy output. It helps to simplify the test logic, to secure alignment in result dtypes, shapes and etc.